openassessment/xblock/openassessmentblock.py

```diff
@@ -746,10 +746,9 @@ def in_studio_preview(self):
             bool
 
         """
-        # When we're running in Studio Preview mode, the XBlock won't provide us with a user ID.
-        # (Note that `self.xmodule_runtime` will still provide an anonymous
-        # student ID, so we can't rely on that)
-        return self.scope_ids.user_id is None
+        # When we're running in Studio Preview mode, the XBlock won't
+        # be tied to a real user
+        return self.xmodule_runtime.get_real_user is None
```

Comment:
Follow the pattern from other instances of calling the `xmodule_runtime`
```suggestion
        if hasattr(self, 'xmodule_runtime'):
            return self.xmodule_runtime.get_real_user is None  # pylint: disable=no-member
        return False
```